PalasoUIWindowsForms.GeckoFxWebBrowserAdapter/WebThumbnailViewer.cs

```diff
@@ -130,17 +144,30 @@ public void LoadItems(IEnumerable<string> pathList)
 				"!function(t,e){\"function\"==typeof define&&define.amd?define(function(){return e(t)}):\"object\"==typeof exports?module.exports=e:t.echo=e(t)}(this,function(t){\"use strict\";var e,n,o,r,c,i={},l=function(){},a=function(t,e){var n=t.getBoundingClientRect();return n.right>=e.l&&n.bottom>=e.t&&n.left<=e.r&&n.top<=e.b},d=function(){(r||!n)&&(clearTimeout(n),n=setTimeout(function(){i.render(),n=null},o))};return i.init=function(n){n=n||{};var a=n.offset||0,u=n.offsetVertical||a,f=n.offsetHorizontal||a,s=function(t,e){return parseInt(t||e,10)};e={t:s(n.offsetTop,u),b:s(n.offsetBottom,u),l:s(n.offsetLeft,f),r:s(n.offsetRight,f)},o=s(n.throttle,250),r=n.debounce!==!1,c=!!n.unload,l=n.callback||l,i.render(),document.addEventListener?(t.addEventListener(\"scroll\",d,!1),t.addEventListener(\"load\",d,!1)):(t.attachEvent(\"onscroll\",d),t.attachEvent(\"onload\",d))},i.render=function(){for(var n,o,r=document.querySelectorAll(\"img[data-echo]\"),d=r.length,u={l:0-e.l,t:0-e.t,b:(t.innerHeight||document.documentElement.clientHeight)+e.b,r:(t.innerWidth||document.documentElement.clientWidth)+e.r},f=0;d>f;f++)o=r[f],a(o,u)?(c&&o.setAttribute(\"data-echo-placeholder\",o.src),o.src=o.getAttribute(\"data-echo\"),c||o.removeAttribute(\"data-echo\"),l(o,\"load\")):c&&(n=o.getAttribute(\"data-echo-placeholder\"))&&(o.src=n,o.removeAttribute(\"data-echo-placeholder\"),l(o,\"unload\"));d||i.detach()},i.detach=function(){document.removeEventListener?t.removeEventListener(\"scroll\",d):t.detachEvent(\"onscroll\",d),clearTimeout(n)},i});");
 			sb.AppendLine("echo.init({offset:200});");
 			sb.AppendLine("</script></body></html>");
+
 			// What we want to do here is _browser.NavigateToString(sb.ToString()).
 			// But the Windows.Forms WebBrowser class doesn't have that method.
-			if (_htmlFile == null)
-				_htmlFile = TempFile.WithExtension("htm");
 			File.WriteAllText(_htmlFile.Path, sb.ToString(), Encoding.UTF8);
 			_browser.Navigate(_htmlFile.Path);
 		}
 
+		//BL-907: Crashes, probably out-of-memory error
+		private string GetThumbnailFile(string fileName)
+		{
+			var cnt = Directory.GetFiles(_tempDirectoryName).Length;
+			var tempFileName = Path.Combine(_tempDirectoryName, "image" + cnt + ".jpg");
```

Comment:
I don't think so, because the directory name is stored in a member variable, so multiple threads would use the same random temp directory. This could lead to both threads creating the same file (for a different image).